exercises/gigasecond/description.md

```diff
@@ -1,3 +1,4 @@
-Calculate the moment when someone has lived for 10^9 seconds.
+Given a `start` value, write a function that returns what the date will
```

Comment:
I'm interested in the fact that this specifies the input, which was otherwise unclear.

If nothing else changes about this exercise, I would like to keep the input clarity change, at the very minimum.

If I must suggest a minimum change that only includes the above, it would be something like:

`Given a starting time as input, determine the moment that is 10^9 seconds past that starting time.`

Reply:
A "starting time" suggests that it must be a Time object, or at least can limit our imagination.  I would like to encourage people to think a little more fundamentally about that.  Giving a single second such as the Integer 1 or even a float of 1.0 should do the expected thing.  Giving a Time object should also work.

However, the thoughts given are from the perspective of Ruby, at least.  If nothing else, I suppose the Ruby track could modify the Readme to indicate this in some way.

Reply:
> A "starting time" suggests that it must be a Time object, or at least can limit our imagination

Without regard to whether I personally agree with this statement or not, I observe the following to be true about this statement:

* I believe that we make our descriptions independent of any particular language, and thus it makes sense to use plain English.
* Regardless of what English sentence someone may suggest, a reader of that sentence might choose to interpret any noun as a possible object
    * Had the suggestion been "Given a starting moment, determine...", a reader might imagine [Moment](https://momentjs.com/) should be a class.
    * Had the suggestion been "Given a starting point, determine..." a reader might imagine PointInTime should be a class
* The English sentence deliberately makes no mention of any specific classes (and it should stay that way), but a person who wishes to port the exercise to a given track might choose to make such an interpretation in consultation with the maintainers of the track.

Next, I note whether I personally agree with the statement

> A "starting time" suggests that it must be a Time object, or at least can limit our imagination

I personally don't agree. The sentence as written doesn't mandate about how a "starting time" is represented in code. If I have a desire to say that an integer represents a starting time, I will think that the sentence as written does not contradict my desire.

> I think that some ambiguity is good, happens in *real life* and helps to provoke discussion about choices as it doesn't limit, unduly, the choices.

I agree. I am thinking the sentence as written meets the requested criterion of not being over-constraining about the choices.

Thus, my suggestion currently remains unchanged.

As a reminder, my suggestion is:

`Given a starting time as input, determine the moment that is 10^9 seconds past that starting time.`

Reply:
I think "moment" is plain English, as it depicts a small segment of time.  The "Gigasecond" implying that the smallest moment is probably a second.  It is defined as "a small (indetefinite) measure of time" and is a well known concept.  A "starting point" is just a more specific indication of what kind of moment.  And I don't believe it needs to be the start of something, it could be measuring from the end of something as well.

I would say, rather, that "Given a starting point" the object may be a StartingPoint class, as a direct translation, but I see your point.

Of course, I fully agree with your third bullet point.

In my statement about the starting time suggesting that it must be, I missed the word "maybe", as in "maybe suggests a Time object".  

I don't think a suggestion is a mandate in any case, ever, so definitely the sentence as written doesn't mandate. But the suggestion may limit the initial thoughts when we see it.  I think that making that decision can be influenced by what we see in the description.  So I also agree with your response.

So I like your suggestion as it is, if we can use "moment" in place of "starting time".  It clearly means the same thing (as implied by addition of a gigasecond) and lets the language communicate the meaning of the word, and the sentence.

Reply:
> So I like your suggestion as it is, if we can use "moment" in place of "starting time". It clearly means the same thing (as implied by addition of a gigasecond) and lets the language communicate the meaning of the word, and the sentence.

OK, no objection here.